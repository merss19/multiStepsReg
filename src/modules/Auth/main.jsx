"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
            t[p[i]] = s[p[i]];
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var react_router_1 = require("react-router");
var react_router_dom_1 = require("react-router-dom");
var ducks_1 = require("./ducks");
var CustomModal_1 = require("modules/CustomModal");
var SignUp_1 = require("modules/SignUp");
var storage_1 = require("tools/storage");
var selectors_1 = require("./selectors");
var mapStateToProps = function (state, ownProps) { return ({
    isFetching: selectors_1.selectIsFetching(state),
    isLoad: selectors_1.selectIsLoad(state),
    isAuth: selectors_1.selectIsAuth(state),
    isError: selectors_1.selectIsError(state),
    errMsg: selectors_1.selectErrMsg(state),
    user: selectors_1.selectUser(state)
}); };
var ProtectedRouteComponent = /** @class */ (function (_super) {
    __extends(ProtectedRouteComponent, _super);
    function ProtectedRouteComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ProtectedRouteComponent.prototype.componentWillMount = function () {
        var _this = this;
        var checkAuth = this.props.checkAuth;
        checkAuth(function (data) { return _this.checkAuth(data); }, function () { return _this.leave(); });
    };
    ProtectedRouteComponent.prototype.componentWillReceiveProps = function (nextProps) {
        if (nextProps.isFetching !== this.props.isFetching ||
            nextProps.isError !== this.props.isError) {
            var isOpen = nextProps.isFetching || nextProps.isError;
            this.props.toggleModal(isOpen);
        }
    };
    ProtectedRouteComponent.prototype.leave = function () {
        var _a = this.props, history = _a.history, toggleModal = _a.toggleModal;
        toggleModal(false);
        storage_1.storage.remove('token', { path: '/' });
        storage_1.storage.remove('txId', { path: '/' });
        storage_1.storage.remove('program', { path: '/' });
        storage_1.storage.remove('packageType', { path: '/' });
        storage_1.storage.remove('promoName', { path: '/' });
        history.replace('/');
    };
    ProtectedRouteComponent.prototype.checkAuth = function (data) {
        var _a = this.props, changeStep = _a.changeStep, history = _a.history;
        if (data.paidState !== 0) {
            storage_1.storage.save('userProgram', data.program, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            if (data.paidState === 2) {
                changeStep(SignUp_1.Steps.four);
                history.replace('/signup');
            }
            else if (data.isFirstEdit) {
                history.replace('/profile');
            }
            else if (!data.emailConfirmed) {
                return null;
            }
            else {
                history.replace('/task');
            }
        }
        else if (!data.program) {
            changeStep(SignUp_1.Steps.two);
        }
        else {
            changeStep(SignUp_1.Steps.four);
        }
    };
    ProtectedRouteComponent.prototype.modalClick = function () {
        var _a = this.props, history = _a.history, toggleModal = _a.toggleModal;
        toggleModal(false);
        history.replace('/');
    };
    ProtectedRouteComponent.prototype.clickHandler = function () {
        var isError = this.props.isError;
        if (!isError) {
            return null;
        }
        return this.modalClick();
    };
    ProtectedRouteComponent.prototype.renderModal = function (modal) {
        var _a = this.props, isFetching = _a.isFetching, errMsg = _a.errMsg, isError = _a.isError;
        return (<CustomModal_1.CustomModal modal={modal} resultText={errMsg} clickHandler={this.clickHandler.bind(this)} isLoader={isFetching}/>);
    };
    ProtectedRouteComponent.prototype.renderContent = function (props) {
        var _a = this.props, ProtectedComponent = _a.component, isFetching = _a.isFetching, errMsg = _a.errMsg;
        return (<div>
        <ProtectedComponent {...props}/>
        {this.renderModal(CustomModal_1.ModalTypes.error)}
      </div>);
    };
    ProtectedRouteComponent.prototype.renderRoute = function (props) {
        var isAuth = this.props.isAuth;
        return (<div>
        {isAuth ?
            this.renderContent(props) :
            this.renderModal(CustomModal_1.ModalTypes.info)}
      </div>);
    };
    ProtectedRouteComponent.prototype.render = function () {
        var _this = this;
        var _a = this.props, ProtectedComponent = _a.component, isAuth = _a.isAuth, isError = _a.isError, isFetching = _a.isFetching, errMsg = _a.errMsg, rest = __rest(_a, ["component", "isAuth", "isError", "isFetching", "errMsg"]);
        return (<react_router_dom_1.Route {...rest} render={function (props) {
            return _this.renderRoute(props);
        }}/>);
    };
    return ProtectedRouteComponent;
}(React.Component));
exports.ProtectedRouteComponent = ProtectedRouteComponent;
exports.ProtectedRoute = react_redux_1.connect(mapStateToProps, {
    checkAuth: ducks_1.checkAuth,
    changeStep: SignUp_1.changeStep,
    toggleModal: CustomModal_1.toggleModal
})(ProtectedRouteComponent);
exports.default = react_router_1.withRouter(exports.ProtectedRoute);
