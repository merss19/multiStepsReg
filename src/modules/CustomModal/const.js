"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ModalTypes;
(function (ModalTypes) {
    ModalTypes["info"] = "info";
    ModalTypes["error"] = "error";
    ModalTypes["success"] = "success";
    ModalTypes["loading"] = "loading";
})(ModalTypes = exports.ModalTypes || (exports.ModalTypes = {}));
