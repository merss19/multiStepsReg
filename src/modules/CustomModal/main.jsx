"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var ReactModal = require("react-modal");
var const_1 = require("./const");
var Button_1 = require("components/Button");
var Loader_1 = require("components/Loader");
var ducks_1 = require("./ducks");
var selectors_1 = require("./selectors");
var react_motion_1 = require("react-motion");
var style = {
    overlay: {
        backgroundColor: 'rgba(45, 43, 43, 0.85)',
        zIndex: '2'
    },
    content: {
        transform: 'translate3d(0%, -50%, 0px)',
        top: '50%',
        padding: '40px',
        maxHeight: '300px',
        maxWidth: '400px',
        overflow: 'none',
        borderRadius: '12px',
        margin: '0 auto',
        textAlign: 'center',
        bottom: 'auto'
    }
};
var mapStateToProps = function (state, ownProps) { return ({
    isOpen: selectors_1.selectIsOpen(state)
}); };
var CustomModalComponent = /** @class */ (function (_super) {
    __extends(CustomModalComponent, _super);
    function CustomModalComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CustomModalComponent.prototype.setBtn = function () {
        var modal = this.props.modal;
        var btn = Button_1.ButtonTypes.info;
        switch (modal) {
            case const_1.ModalTypes.success:
                btn = Button_1.ButtonTypes.success;
                break;
            case const_1.ModalTypes.error:
                btn = Button_1.ButtonTypes.error;
                break;
            case const_1.ModalTypes.info:
                btn = Button_1.ButtonTypes.info;
                break;
            default:
                btn = Button_1.ButtonTypes.info;
        }
        return btn;
    };
    CustomModalComponent.prototype.renderContent = function () {
        var _this = this;
        var _a = this.props, resultText = _a.resultText, isLoader = _a.isLoader;
        var btn = this.setBtn();
        if (isLoader) {
            return <Loader_1.Loader />;
        }
        else {
            return (<react_motion_1.Motion defaultStyle={{ opacity: 0 }} style={{ opacity: react_motion_1.spring(1, { stiffness: 70 }) }}>
          {function (interpolatedStyle) {
                return (<div className='modal' style={__assign({}, interpolatedStyle)}>
                  <h2 className='modal__title'>{resultText}</h2>
                  <Button_1.Button onClick={_this.closeModal.bind(_this)} styleBtn={btn}>
                    Продолжить
                  </Button_1.Button>
                </div>);
            }}
        </react_motion_1.Motion>);
        }
    };
    CustomModalComponent.prototype.closeModal = function () {
        var _a = this.props, toggleModal = _a.toggleModal, clickHandler = _a.clickHandler;
        toggleModal(false);
        if (!clickHandler) {
            return;
        }
        clickHandler();
    };
    CustomModalComponent.prototype.render = function () {
        var isOpen = this.props.isOpen;
        return (<div>
        <ReactModal isOpen={isOpen} style={style} contentLabel='Action Modal' onRequestClose={this.closeModal.bind(this)}>
          <div className='modal__content'>
            {this.renderContent()}
          </div>

        </ReactModal>
      </div>);
    };
    CustomModalComponent.defaultProps = {
        isLoader: false
    };
    return CustomModalComponent;
}(React.PureComponent));
exports.CustomModalComponent = CustomModalComponent;
exports.CustomModal = react_redux_1.connect(mapStateToProps, { toggleModal: ducks_1.toggleModal })(CustomModalComponent);
exports.default = exports.CustomModal;
