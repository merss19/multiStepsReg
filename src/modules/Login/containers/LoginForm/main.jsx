"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var redux_form_1 = require("redux-form");
var react_router_dom_1 = require("react-router-dom");
var InputText_1 = require("components/InputText");
var _1 = require("components/Button/");
var formValidations = require("tools/validations");
var LoginForm = /** @class */ (function (_super) {
    __extends(LoginForm, _super);
    function LoginForm() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    LoginForm.prototype.render = function () {
        var _a = this.props, handleSubmit = _a.handleSubmit, onSubmit = _a.onSubmit;
        return (<div className='entry entry--min'>
        <div className='entry__inner'>

          <div className='entry__header'>
            <h2 className='entry__title text-center'>Вход в Личный кабинет</h2>
          </div>

          <form onSubmit={handleSubmit(onSubmit)} className='entry__box'>
            <redux_form_1.Field name='email' placeholder='Ваш e-mail' type='email' component={InputText_1.default}/>
            <redux_form_1.Field name='password' placeholder='Ваш пароль' type='password' component={InputText_1.default}/>

            <_1.Button type='submit' styleBtn={_1.ButtonTypes.info} wide={true}>
              Войти
            </_1.Button>

            <ul className='entry-nav mtb20'>
              <li className='entry-nav__item'>
                <react_router_dom_1.Link to='/signup'>Регистрация</react_router_dom_1.Link>
              </li>
              <li className='entry-nav__item'>
                <react_router_dom_1.Link to='/restore'>Забыли пароль?</react_router_dom_1.Link>
              </li>
            </ul>

          </form>
          {this.props.children}
        </div>
      </div>);
    };
    return LoginForm;
}(React.PureComponent));
exports.LoginForm = LoginForm;
exports.default = redux_form_1.reduxForm({
    form: 'LoginFormm',
    validate: formValidations.validateLogin
})(LoginForm);
