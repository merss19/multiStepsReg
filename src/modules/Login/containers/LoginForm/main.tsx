import * as React from 'react';
import {Field, FormProps, reduxForm} from 'redux-form';
import {Link} from 'react-router-dom';
import InputText from 'components/InputText';
import {Button, ButtonTypes} from 'components/Button/';
import * as formValidations from 'tools/validations';

interface LoginFormData {
  email: string
  password: string
}

interface OwnProps extends FormProps<LoginFormData, {}, {}> {
  onSubmit: (values: LoginFormData) => void
}

export class LoginForm extends React.PureComponent<OwnProps, {}> {
  render() {
    const {handleSubmit, onSubmit} = this.props;
    return (
      <div className='entry entry--min'>
        <div className='entry__inner'>

          <div className='entry__header'>
            <h2 className='entry__title text-center'>Вход в Личный кабинет</h2>
          </div>

          <form onSubmit={handleSubmit(onSubmit)} className='entry__box'>
            <Field name='email'
                   placeholder='Ваш e-mail'
                   type='email'
                   component={InputText}
            />
            <Field name='password'
                   placeholder='Ваш пароль'
                   type='password'
                   component={InputText}
            />

            <Button
              type='submit'
              styleBtn={ButtonTypes.info}
              wide={true}
            >
              Войти
            </Button>

            <ul className='entry-nav mtb20'>
              <li className='entry-nav__item'>
                <Link to='/signup'>Регистрация</Link>
              </li>
              <li className='entry-nav__item'>
                <Link to='/restore'>Забыли пароль?</Link>
              </li>
            </ul>

          </form>
          {this.props.children}
        </div>
      </div>
    );
  }
}

export default reduxForm({
  form: 'LoginFormm',
  validate: formValidations.validateLogin
})(LoginForm);
