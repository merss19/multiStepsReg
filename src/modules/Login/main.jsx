"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var LoginForm_1 = require("./containers/LoginForm");
var ducks_1 = require("./ducks");
var react_router_1 = require("react-router");
var CustomModal_1 = require("../CustomModal");
var LayoutHOC_1 = require("components/HOC/LayoutHOC");
var storage_1 = require("tools/storage");
var SignUp_1 = require("../SignUp");
var selectors_1 = require("./selectors");
var mapStateToProps = function (state) { return ({
    isFetching: selectors_1.selectIsFetching(state),
    isLoad: selectors_1.selectIsLoad(state),
    isError: selectors_1.selectIsError(state),
    errMsg: selectors_1.selectErrMsg(state),
    profile: selectors_1.selectProfile(state)
}); };
var LoginComponent = /** @class */ (function (_super) {
    __extends(LoginComponent, _super);
    function LoginComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    LoginComponent.prototype.componentWillReceiveProps = function (nextProps) {
        if (nextProps.isFetching !== this.props.isFetching || nextProps.isError !== this.props.isError) {
            var isOpen = nextProps.isFetching || nextProps.isError;
            this.props.toggleModal(isOpen);
        }
    };
    LoginComponent.prototype.onSubmit = function (data) {
        var _this = this;
        this.props.submitLogin(data, function (data) { return _this.submitCb(data); });
    };
    LoginComponent.prototype.submitCb = function (data) {
        var _a = this.props, changeStep = _a.changeStep, setToken = _a.setToken, history = _a.history;
        if (data && data.paidState !== 0) {
            setToken(data.authToken);
            storage_1.storage.save('token', data.authToken, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            storage_1.storage.save('userProgram', data.program, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            if (data.paidState === 2) {
                changeStep(SignUp_1.Steps.four);
            }
            else if (data.isFirstEdit) {
                history.replace('/profile');
            }
            else {
                history.replace('/task');
            }
        }
        else if (!data.program) {
            storage_1.storage.save('token', data.authToken, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            changeStep(SignUp_1.Steps.two);
            history.replace('/signup');
        }
        else {
            storage_1.storage.save('token', data.authToken, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            changeStep(SignUp_1.Steps.three);
            history.replace('/signup');
        }
    };
    LoginComponent.prototype.setError = function () {
        this.props.setError(false);
    };
    LoginComponent.prototype.render = function () {
        var _a = this.props, isFetching = _a.isFetching, errMsg = _a.errMsg;
        return (<LoginForm_1.LoginForm onSubmit={this.onSubmit.bind(this)}>
        <CustomModal_1.CustomModal modal={CustomModal_1.ModalTypes.error} resultText={errMsg} isLoader={isFetching} clickHandler={this.setError.bind(this)}/>
      </LoginForm_1.LoginForm>);
    };
    return LoginComponent;
}(React.Component));
exports.LoginComponent = LoginComponent;
exports.Login = react_redux_1.connect(mapStateToProps, {
    submitLogin: ducks_1.submitLogin,
    changeStep: SignUp_1.changeStep,
    setToken: ducks_1.setToken,
    setError: ducks_1.setError,
    toggleModal: CustomModal_1.toggleModal
})(LoginComponent);
exports.default = LayoutHOC_1.LayoutHOC()(react_router_1.withRouter(exports.Login));
