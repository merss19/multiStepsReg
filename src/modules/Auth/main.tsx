import * as React from 'react';
import {connect} from 'react-redux';
import {RouteProps, RouteComponentProps, withRouter} from 'react-router';
import {Route} from 'react-router-dom';
import {checkAuth} from './ducks'
import {CustomModal, toggleModal, ModalTypes, ModalType} from 'modules/CustomModal';
import {changeStep, Steps} from 'modules/SignUp';
import {storage} from 'tools/storage';
import {
  selectIsFetching,
  selectIsLoad,
  selectUser,
  selectIsAuth,
  selectErrMsg,
  selectIsError
} from './selectors'

type ConnectedDispatch = {
  checkAuth(cb: (data?: any) => void, cbFail: () => void): void
  toggleModal(data: boolean): void
  changeStep: (data: Steps) => void
}

interface OwnProps extends RouteProps {
}

type ConnectedState = {
  isFetching: boolean
  isLoad: boolean
  isError: boolean
  isAuth: boolean
  errMsg: string
  user: any
}

const mapStateToProps = (state: any, ownProps: OwnProps): ConnectedState => ({
  isFetching: selectIsFetching(state),
  isLoad: selectIsLoad(state),
  isAuth: selectIsAuth(state),
  isError: selectIsError(state),
  errMsg: selectErrMsg(state),
  user: selectUser(state)
});
type Props = RouteComponentProps<string> & OwnProps & ConnectedDispatch & ConnectedState;

export class ProtectedRouteComponent extends React.Component<Props, {}> {

  componentWillMount() {
    const {checkAuth} = this.props;
    checkAuth((data) => this.checkAuth(data), () => this.leave());
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.isFetching !== this.props.isFetching ||
        nextProps.isError !== this.props.isError) {
      const isOpen: boolean = nextProps.isFetching || nextProps.isError;
      this.props.toggleModal(isOpen);
    }
  }

  leave() {
    const {history, toggleModal} = this.props;
    toggleModal(false);
    storage.remove('token', {path: '/'});
    storage.remove('txId', {path: '/'});
    storage.remove('program', {path: '/'});
    storage.remove('packageType', {path: '/'});
    storage.remove('promoName', {path: '/'});
    history.replace('/');
  }

  checkAuth(data: any) : void {
    const {changeStep, history} = this.props;
    if (data.paidState !== 0) {
      storage.save('userProgram', data.program, {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      if (data.paidState === 2) {
        changeStep(Steps.four);
        history.replace('/signup');
      } else if (data.isFirstEdit) {
        history.replace('/profile');
      } else if (!data.emailConfirmed) {
        return null;
      } else {
        history.replace('/task');
      }
    } else if (!data.program) {
      changeStep(Steps.two);
    } else {
      changeStep(Steps.four);
    }
  }

  modalClick(): void {
    const {history, toggleModal} = this.props;
    toggleModal(false);
    history.replace('/');
  }

  clickHandler(){
    const {isError} = this.props
    if(!isError ){
      return null
    }
    return this.modalClick()
  }


  renderModal(modal:ModalType){
    const {isFetching, errMsg, isError} = this.props
    return(
      <CustomModal
        modal={modal}
        resultText={errMsg}
        clickHandler={this.clickHandler.bind(this)}
        isLoader={isFetching}
      />)
  }

  renderContent(props: any){
    const {component: ProtectedComponent, isFetching, errMsg} = this.props
    return(
      <div>
        <ProtectedComponent {...props}/>
        {this.renderModal(ModalTypes.error)}
      </div>)
  }

  renderRoute(props: any){
    const {isAuth} = this.props;
    return(
      <div>
        {isAuth ?
          this.renderContent(props) :
          this.renderModal(ModalTypes.info)
        }
      </div>)
  }

  render() {
    const {component: ProtectedComponent, isAuth, isError, isFetching, errMsg, ...rest} = this.props;
    return (
      <Route {...rest} render={props => {
        return this.renderRoute(props);
      }
      }/>
      )
  }
}


export const ProtectedRoute = connect(
  mapStateToProps,
  {
    checkAuth,
    changeStep,
    toggleModal
  })(ProtectedRouteComponent)


export default withRouter(ProtectedRoute);

