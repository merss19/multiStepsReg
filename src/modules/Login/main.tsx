import * as React from 'react';
import {connect} from 'react-redux';
import {LoginForm} from './containers/LoginForm';
import {submitLogin, setToken, setError} from './ducks';
import {LoginFormData} from './interfaces';
import {withRouter, RouteComponentProps} from 'react-router'
import {CustomModal, toggleModal, ModalTypes} from '../CustomModal';
import {LayoutHOC} from 'components/HOC/LayoutHOC';
import {storage} from 'tools/storage';
import {changeStep, Steps} from '../SignUp';


import {
  selectIsFetching,
  selectIsLoad,
  selectProfile,
  selectErrMsg,
  selectIsError
} from './selectors'

type ConnectedDispatch = {
  changeStep: (data: Steps) => void
  submitLogin(data: LoginFormData, cb: (data: any) => void): void
  toggleModal(data: boolean): void
  setToken(toket: string): void
  setError(data: boolean): any
}
type ConnectedState = {
  isFetching: boolean
  isLoad: boolean
  isError: boolean
  errMsg: string,
  profile: any
}
const mapStateToProps = (state: any): ConnectedState => ({
  isFetching: selectIsFetching(state),
  isLoad: selectIsLoad(state),
  isError: selectIsError(state),
  errMsg: selectErrMsg(state),
  profile: selectProfile(state)
});

type props = RouteComponentProps<string> & ConnectedDispatch & ConnectedState;

export class LoginComponent extends React.Component<props, any> {

  componentWillReceiveProps(nextProps: props) {
    if (nextProps.isFetching !== this.props.isFetching || nextProps.isError !== this.props.isError) {
      const isOpen: boolean = nextProps.isFetching || nextProps.isError;
      this.props.toggleModal(isOpen);
    }
  }
  onSubmit(data: LoginFormData) {
    this.props.submitLogin(data, (data) => this.submitCb(data));
  }

  submitCb(data: any): void {
    const {changeStep, setToken, history} = this.props;
    if (data && data.paidState !== 0) {
      setToken(data.authToken);
      storage.save('token', data.authToken, {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      storage.save(
        'userProgram', data.program, {path: '/', maxAge: 60 * 60 * 24 * 365 * 10}
      );
      if (data.paidState === 2) {
        changeStep(Steps.four);
      } else if (data.isFirstEdit) {
        history.replace('/profile');
      } else {
        history.replace('/task');
      }
    } else if (!data.program) {
      storage.save('token', data.authToken, {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      changeStep(Steps.two);
      history.replace('/signup');
    } else {
      storage.save('token', data.authToken, {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      changeStep(Steps.three);
      history.replace('/signup');
    }
  }

  setError(){
    this.props.setError(false)
  }

  render() {
    const {isFetching, errMsg} = this.props;
    return (
      <LoginForm onSubmit={this.onSubmit.bind(this)}>
        <CustomModal
          modal={ModalTypes.error}
          resultText={errMsg}
          isLoader={isFetching}
          clickHandler={this.setError.bind(this)}
        />
      </LoginForm>
    );
  }
}

export const Login = connect(mapStateToProps, {
  submitLogin,
  changeStep,
  setToken,
  setError,
  toggleModal
})(LoginComponent);
export default LayoutHOC()(withRouter(Login));