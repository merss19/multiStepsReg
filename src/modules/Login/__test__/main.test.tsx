import {shallow, mount} from 'enzyme';
import {Login, LoginComponent} from '../main';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import * as sinon from 'sinon';
import {Provider} from 'react-redux';
import configureStore from '../../../store';
import {BrowserRouter as Router} from 'react-router-dom';
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {Steps} from "../../SignUp/const";
import {storage} from "../../../tools/storage/index";

const store = configureStore();
configure({adapter: new Adapter()});

let data = {
  paidState: 2,
  program: 11,
  isFirstEdit: false,
  authToken: 'qweqwe'
};
let payload = {
  email: 'test@email.ru',
  password: 'qweqwe'
};
describe('Component LoginForm', () => {
  let changeStep,
    submitLogin,
    toggleModal,
    setToken,
    setError,
    isFetching,
    isLoad,
    isError,
    errMsg,
    profile,
    match,
    location,
    history;
  beforeEach(() => {
    changeStep = jest.fn(),
      submitLogin = jest.fn(),
      toggleModal = jest.fn(),
      setToken = jest.fn(),
      setError = jest.fn(),
      isFetching = false,
      isLoad = false,
      isError = false,
      errMsg = 'Ошибка',
      profile = {},
      match = {},
      location = {},
      history = {
        replace: jest.fn()
      }
  });
  const buildComponent = () => {
    const props = {
      changeStep: changeStep,
      submitLogin: submitLogin,
      toggleModal: toggleModal,
      setToken: setToken,
      setError: setError,
      isFetching: isFetching,
      isLoad: isLoad,
      isError: isError,
      errMsg: errMsg,
      profile: profile,
      match: match,
      location: location,
      history: history
    };
    return shallow(<LoginComponent {...props}/>)
  };
  const buildComponentMount = () => {
    const props = {
      changeStep: changeStep,
      submitLogin: submitLogin,
      toggleModal: toggleModal,
      isFetching: isFetching,
      isLoad: isLoad,
      isError: isError,
      profile: profile
    };
    return mount(
      <Provider store={store}>
        <Router>
          <Login {...props}/>
        </Router>
      </Provider>)
  };
  const spy = sinon.spy(Login.prototype, 'componentWillReceiveProps');
  it('renders component', () => {
    const component = buildComponent()
    expect(component).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = toJson(buildComponent());
    expect(tree).toMatchSnapshot();
  });
  it('renders component Mount', () => {
    const component = buildComponentMount();
    expect(component).toHaveLength(1);
  });
  it('do not call toggleModal', () => {
    const component = buildComponent();
    expect(toggleModal).not.toHaveBeenCalled()
    isLoad = true;
    isFetching = false;
    component.setProps({isLoad:isLoad, isFetching:isFetching});
    expect(toggleModal).not.toHaveBeenCalled();
  });
  it('call toggleModal isFetching = true', () => {
    const component = buildComponent();
    isFetching = true;
    component.setProps({isFetching: isFetching});
    expect(toggleModal).toHaveBeenCalledWith(true);
  });
  it('call toggleModal isError = true', () => {
    const component = buildComponent();
    isError = true;
    component.setProps({isError: isError});
    expect(toggleModal).toHaveBeenCalledWith(true)
  });
  it('call setError', () => {
    const component = buildComponent();
    const instance = component.instance() as LoginComponent;
    instance.setError();
    expect(setError).toHaveBeenCalledWith(false);
  });
  it('call submitCb - call submitLogin', () => {
    const component = buildComponent();
    const instance = component.instance() as LoginComponent;
    instance.onSubmit(payload);
    expect(submitLogin).toHaveBeenCalled();
  });
  it('call submitCb, paidState !== 0', () => {
    const component = buildComponent();
    const instance = component.instance() as LoginComponent;
    instance.submitCb(data);
    expect(changeStep).toHaveBeenCalledWith(Steps.four);
    expect(setToken).toHaveBeenCalledWith(data.authToken);
    expect(storage.load('token')).toBe(data.authToken);
    expect(parseInt(storage.load('userProgram'))).toBe(data.program);
  });
  it('call submitCb, paidState === 2', () => {
    const component = buildComponent();
    const instance = component.instance() as LoginComponent;
    instance.submitCb(data);
    expect(changeStep).toHaveBeenCalledWith(Steps.four);
  });
  it('call submitCb, paidState !== 2', () => {
    const component = buildComponent();
    const instance = component.instance() as LoginComponent;
    data.paidState = 1;
    instance.submitCb(data);
    expect(history.replace).toHaveBeenCalledWith('/task');
  });
  it('call submitCb, isFirstEdit === true', () => {
    const component = buildComponent();
    const instance = component.instance() as LoginComponent;
    data.isFirstEdit = true;
    instance.submitCb(data);
    expect(history.replace).toHaveBeenCalledWith('/profile');
  });
  it('call submitCb, isFirstEdit !== true', () => {
    const component = buildComponent();
    const instance = component.instance() as LoginComponent;
    data.isFirstEdit = false;
    instance.submitCb(data);
    expect(history.replace).toHaveBeenCalledWith('/task');
  });
  it('call submitCb, !data.program', () => {
    const component = buildComponent();
    const instance = component.instance() as LoginComponent;
    data.paidState = 0;
    data.program = null;
    instance.submitCb(data);
    expect(history.replace).toHaveBeenCalledWith('/signup');
    expect(changeStep).toHaveBeenCalledWith(Steps.two);
    expect(storage.load('token')).toBe(data.authToken);
  });
  it('call submitCb, paidState === 0', () => {
    const component = buildComponent();
    const instance = component.instance() as LoginComponent;
    data.paidState = 0;
    data.isFirstEdit = true;
    data.program = 11;
    instance.submitCb(data);
    expect(history.replace).toHaveBeenCalledWith('/signup');
    expect(changeStep).toHaveBeenCalledWith(Steps.three);
    expect(storage.load('token')).toBe(data.authToken);
  })
});


