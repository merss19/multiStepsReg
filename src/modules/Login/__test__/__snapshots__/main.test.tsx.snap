// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LoginForm renders correctly 1`] = `
<ReduxForm
  onSubmit={[Function]}
>
  <Connect(CustomModalComponent)
    clickHandler={[Function]}
    isLoader={false}
    modal="error"
    resultText="Ошибка"
  />
</ReduxForm>
`;
